py3status/modules/net_rate.py

```diff
@@ -123,16 +141,44 @@ def currentSpeed(self):
             interface = None
             hide = self.hide_if_zero
 
-        return {
-            'cached_until': self.py3.time_in(self.cache_timeout),
-            'full_text': "" if hide else
-            self.py3.safe_format(self.format,
-                                 dict(total=self._divide_and_format(delta['total']),
-                                      up=self._divide_and_format(delta['up']),
-                                      down=self._divide_and_format(delta['down']),
-                                      interface=interface[:-1])
-                                 ) if interface else self.format_no_connection
-        }
+        if interface:
+
+            if delta['down'] < self.threshold_slow:
+                self.py3.COLOR_DOWN = self.py3.COLOR_BAD
+            elif delta['down'] < self.threshold_fast:
+                self.py3.COLOR_DOWN = self.py3.COLOR_DEGRADED
+            else:
+                self.py3.COLOR_DOWN = self.py3.COLOR_GOOD
+
+            if delta['total'] < self.threshold_slow:
+                self.py3.COLOR_TOTAL = self.py3.COLOR_BAD
+            elif delta['total'] < self.threshold_fast:
+                self.py3.COLOR_TOTAL = self.py3.COLOR_DEGRADED
+            else:
+                self.py3.COLOR_TOTAL = self.py3.COLOR_GOOD
+
+            if delta['up'] < self.threshold_slow:
+                self.py3.COLOR_UP = self.py3.COLOR_BAD
+            elif delta['up'] < self.threshold_fast:
+                self.py3.COLOR_UP = self.py3.COLOR_DEGRADED
+            else:
+                self.py3.COLOR_UP = self.py3.COLOR_GOOD
```

Comment:
Ok. I thought that was the idea, but obviously I'm wrong.

if someone specifies `[\?color=up text]`, then the color of the text should be the color of `up`, but this varies with threshold, no? We could maybe mix colors and conditionals, but it may get ugly: `[\?critical[\?color=up_crit text]][\?degraded[\?color=up_degr text]]...`